src/opm/output/eclipse/Summary.cpp

```diff
@@ -592,6 +592,101 @@ inline quantity spr ( const fn_args& args ) {
+
+    if( args.schedule_wells.empty() ) return zero;
+    // Like completion rate we need to look
+    // up a connection with offset 0.
+    const size_t segNumber = args.num;
+    if( args.schedule_wells.empty() ) return zero;
+
+    const auto& well = args.schedule_wells.front();
+    const auto& name = well.name();
+    if( args.wells.count( name ) == 0 ) return zero;
+
+    const auto& well_data = args.wells.at( name );
+
+    const auto& segment = well_data.segments.find(segNumber);
+
+    if( segment == well_data.segments.end() ) return zero;
+
+
+    const auto& v = segment->second.pressure_drop;
+    return { v, measure::pressure };
+}
+
+inline quantity sprdh ( const fn_args& args ) {
+    const quantity zero = { 0, measure::pressure };
+
+    if( args.schedule_wells.empty() ) return zero;
+    // Like completion rate we need to look
+    // up a connection with offset 0.
+    const size_t segNumber = args.num;
+    if( args.schedule_wells.empty() ) return zero;
+
+    const auto& well = args.schedule_wells.front();
+    const auto& name = well.name();
+    if( args.wells.count( name ) == 0 ) return zero;
+
+    const auto& well_data = args.wells.at( name );
+
+    const auto& segment = well_data.segments.find(segNumber);
+
+    if( segment == well_data.segments.end() ) return zero;
+
+
+    const auto& v = segment->second.pressure_drop_hydrostatic;
+    return { v, measure::pressure };
+}
+
+inline quantity sprdf ( const fn_args& args ) {
+    const quantity zero = { 0, measure::pressure };
+
+    if( args.schedule_wells.empty() ) return zero;
+    // Like completion rate we need to look
+    // up a connection with offset 0.
+    const size_t segNumber = args.num;
+    if( args.schedule_wells.empty() ) return zero;
+
+    const auto& well = args.schedule_wells.front();
+    const auto& name = well.name();
+    if( args.wells.count( name ) == 0 ) return zero;
+
+    const auto& well_data = args.wells.at( name );
+
+    const auto& segment = well_data.segments.find(segNumber);
+
+    if( segment == well_data.segments.end() ) return zero;
+
+
+    const auto& v = segment->second.pressure_drop_friction;
+    return { v, measure::pressure };
+}
+
+inline quantity sprda ( const fn_args& args ) {
+    const quantity zero = { 0, measure::pressure };
+
+    if( args.schedule_wells.empty() ) return zero;
+    // Like completion rate we need to look
+    // up a connection with offset 0.
+    const size_t segNumber = args.num;
+    if( args.schedule_wells.empty() ) return zero;
+
+    const auto& well = args.schedule_wells.front();
+    const auto& name = well.name();
+    if( args.wells.count( name ) == 0 ) return zero;
+
+    const auto& well_data = args.wells.at( name );
+
+    const auto& segment = well_data.segments.find(segNumber);
+
+    if( segment == well_data.segments.end() ) return zero;
+
+
+    const auto& v = segment->second.pressure_drop_acceleration;
+    return { v, measure::pressure };
+}
```

Comment:
It must be possible to refactor this to have only one `sprxx` function - now you have four 99% identical functions.

Reply:
this is fixed when addressing the comment above.